docker/Dockerfile

```diff
@@ -1,27 +1,22 @@
-FROM ubuntu:xenial
+FROM python:3-slim
 
-MAINTAINER Herman Junge <herman.junge@consensys.net>
+ARG SOLC_URL="https://github.com/ethereum/solidity/releases/download/v0.4.25/solc-static-linux"
 
-RUN apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 923F6CA9 \
-    && echo "deb http://ppa.launchpad.net/ethereum/ethereum/ubuntu xenial main" \
-       | tee -a /etc/apt/sources.list.d/ethereum.list
+ADD $SOLC_URL /usr/local/bin
 
-RUN apt-get update -y \
-    && apt-get dist-upgrade -y
+RUN chmod +x /usr/local/bin/solc-static-linux \
+    && mv /usr/local/bin/solc-static-linux /usr/local/bin/solc
 
-RUN apt-get install -y automake build-essential git-core libffi-dev \
-       libgmp-dev libssl-dev libtool pkg-config python-dev python-pip solc \
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends build-essential git-core \
     && rm -rf /var/lib/apt/lists/*
 
-RUN pip install --upgrade pip
+RUN git clone https://github.com/raiden-network/raiden.git /app/raiden \
```

Comment:
After `git clone`, please `rm -rf /app/raiden/.git`, it's quite big and surely not needed in the final image